modules/internal/ChapelRange.chpl

```diff
@@ -144,7 +144,8 @@ module ChapelRange {
   
     var _low: idxType = 1;                         // lower bound
     var _high: idxType = 0;                        // upper bound
-    var _stride: strType = 1;                      // signed stride of range
+    var _stride: if stridable then strType else void =
+                   if stridable then 1 else _void; // signed stride of range
     var _alignment: idxType = 0;                   // alignment
     var _aligned : bool = false;
```

Comment:
Ultimately, I'd really like alignment and aligned to also be void in the event that stridable is false.  But I think that can/should wait for another PR (arguably perhaps even the stride change should be, but I'm not going to object to it being here as well...).